qa/integration-tests/src/test/java/io/camunda/it/auth/DecisionAuthorizationIT.java

```diff
@@ -0,0 +1,237 @@
+    assertThat(decisionDefinitions.stream().map(p -> p.getDmnDecisionId()).toList())
+        .containsOnly(DECISION_DEFINITION_ID_1);
+  }
+
+  @TestTemplate
+  void shouldReturnForbiddenForUnauthorizedDecisionDefinition(
+      @Authenticated(ADMIN) final ZeebeClient adminClient,
+      @Authenticated(RESTRICTED) final ZeebeClient userClient) {
+    // given
+    final var decisionDefinitionKey =
+        getDecisionDefinitionKey(adminClient, DECISION_DEFINITION_ID_2);
+
+    // when
+    final Executable executeGet =
+        () -> userClient.newDecisionDefinitionGetRequest(decisionDefinitionKey).send().join();
+
+    // then
+    final var problemException = assertThrows(ProblemException.class, executeGet);
+    assertThat(problemException.code()).isEqualTo(403);
+    assertThat(problemException.details().getDetail())
+        .isEqualTo("Unauthorized to perform operation 'READ' on resource 'DECISION_DEFINITION'");
+  }
+
+  @TestTemplate
+  void shouldReturnAuthorizedDecisionDefinition(
+      @Authenticated(ADMIN) final ZeebeClient adminClient,
+      @Authenticated(RESTRICTED) final ZeebeClient userClient) {
+    // given
+    final var decisionDefinitionKey =
+        getDecisionDefinitionKey(adminClient, DECISION_DEFINITION_ID_1);
+
+    // when
+    final var decisionDefinition =
+        userClient.newDecisionDefinitionGetRequest(decisionDefinitionKey).send().join();
+
+    // then
+    assertThat(decisionDefinition).isNotNull();
+    assertThat(decisionDefinition.getDmnDecisionId()).isEqualTo(DECISION_DEFINITION_ID_1);
+  }
+
+  @TestTemplate
+  void shouldReturnAuthorizedDecisionRequirements(
+      @Authenticated(RESTRICTED) final ZeebeClient userClient) {
+    // when
+    final var decisionRequirements =
+        userClient.newDecisionRequirementsQuery().send().join().items();
+
+    // then
+    assertThat(decisionRequirements).hasSize(1);
+    assertThat(decisionRequirements.stream().map(p -> p.getDmnDecisionRequirementsId()).toList())
+        .containsOnly(DECISION_REQUIREMENTS_ID_1);
+  }
+
+  @TestTemplate
+  void shouldReturnForbiddenForUnauthorizedDecisionRequirements(
+      @Authenticated(ADMIN) final ZeebeClient adminClient,
+      @Authenticated(RESTRICTED) final ZeebeClient userClient) {
+    // given
+    final var decisionRequirementsKey =
+        getDecisionRequirementsKey(adminClient, DECISION_REQUIREMENTS_ID_2);
+
+    // when
+    final Executable executeGet =
+        () -> userClient.newDecisionRequirementsGetRequest(decisionRequirementsKey).send().join();
+
+    // then
+    final var problemException = assertThrows(ProblemException.class, executeGet);
+    assertThat(problemException.code()).isEqualTo(403);
+    assertThat(problemException.details().getDetail())
+        .isEqualTo(
+            "Unauthorized to perform operation 'READ' on resource 'DECISION_REQUIREMENTS_DEFINITION'");
+  }
+
+  @TestTemplate
+  void shouldReturnAuthorizedDecisionRequirements(
```

Comment:
🙃 This name `shouldReturnAuthorizedDecisionRequirements` already exists a few lines above. It only works because it takes different arguments. Maybe we can rename this one to reflect it's `getByKey` while the other one is `search`.